components/camel-soroush/src/main/java/org/apache/camel/component/soroushbot/component/SoroushBotAbstractConsumer.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.component.soroushbot.component;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.component.soroushbot.models.Endpoint;
+import org.apache.camel.component.soroushbot.models.SoroushMessage;
+import org.apache.camel.component.soroushbot.service.SoroushService;
+import org.apache.camel.support.DefaultConsumer;
+import org.glassfish.jersey.client.ClientProperties;
+import org.glassfish.jersey.media.sse.EventInput;
+import org.glassfish.jersey.media.sse.InboundEvent;
+import org.glassfish.jersey.media.sse.SseFeature;
+
+import static org.apache.camel.component.soroushbot.utils.StringUtils.ordinal;
+
+/**
+ * this component handle logic for getting message from Soroush server and for each message
+ * it calls abstract function {@link SoroushBotAbstractConsumer#sendExchange(Exchange)}
+ * each subclass should handle how it will start the processing of the exchange
+ */
+public abstract class SoroushBotAbstractConsumer extends DefaultConsumer {
+    SoroushBotEndpoint endpoint;
+    /**
+     * {@link ObjectMapper} for parse message JSON
+     */
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    public SoroushBotAbstractConsumer(SoroushBotEndpoint endpoint, Processor processor) {
+        super(endpoint, processor);
+        this.endpoint = endpoint;
+    }
+
+    /**
+     * create an {@link EventInput} that connect to soroush SSE server and read events
+     *
+     * @param target
+     * @return
+     */
+    private EventInput createEvent(WebTarget target) {
+        EventInput eventInput = target.request().get(EventInput.class);
+        eventInput.setChunkType(MediaType.SERVER_SENT_EVENTS);
+        return eventInput;
+    }
+
+    @Override
+    public void doStart() {
+//     create new Thread for listening to Soroush SSE Server so that it release the main camel thread.
+        Thread thread = new Thread(() -> {
```

Comment:
Should use Camel's ExecutorServiceStrategy to create a single thread instead, which you can get via the camel context